examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,396 @@
+    estimator = RidgeCV()
+    estimator.fit(X, y)
+    y_predict = estimator.predict(X)
+
+    # Save the beta estimates, the variance of these estimates and 1.96 *
+    # standard error of the estimates. The latter is useful to know the 95%
+    # confidence interval.
+    full_coefs = estimator.coef_
+    full_vars = calculate_variance_of_beta_estimates(y, y_predict, X)
+
+    return full_coefs, full_vars
+
+
+def get_results_chained_imputation(X_incomplete, y, random_state=0):
+    # Impute incomplete data with IterativeImputer using single imputation
+    # We perform N_ITER imputations and only use the last imputation.
+    imputer = IterativeImputer(n_iter=N_ITER,
+                               sample_posterior=True,
+                               random_state=random_state)
+    imputer.fit(X_incomplete)
+    X_imputed = imputer.transform(X_incomplete)
+
+    # Perform linear regression on chained single imputed data
+    # Estimate beta estimates and their variances
+    estimator = RidgeCV()
+    estimator.fit(X_imputed, y)
+    y_predict = estimator.predict(X_imputed)
+
+    # Save the beta estimates, the variance of these estimates
+    chained_coefs = estimator.coef_
+    chained_vars = calculate_variance_of_beta_estimates(
+            y, y_predict, X_imputed)
+
+    return chained_coefs, chained_vars
+
+
+def get_results_mice_imputation(X_incomplete, y, num_imputations=5):
+    # Perform a model on each of the m imputed datasets
+    # Estimate the estimates for each model/dataset
+    m_coefs = []
+    m_vars = []
+    for i in range(num_imputations):
+        m_coef, m_var = get_results_chained_imputation(X_incomplete, y,
+                                                       random_state=i)
+        m_coefs.append(m_coef)
+        m_vars.append(m_var)
+
+    # Calculate the end estimates by applying Rubin's rules.
+    Qbar, T = rubins_pooling_rules(m_coefs, m_vars)
+
+    return Qbar, T
+
+
+# The original multiple imputation procedure as developed under the name
+# MICE includes all variables in the imputation process; including the output
+# variable. The reason to do this is that the imputation model should at least
+# contain the analysis model to result in unbiased estimates. In this function,
+# we will also include y in the imputation process.
+def get_results_mice_imputation_includingy(X_incomplete, y, num_imputations=5):
+    Xy = np.column_stack((X_incomplete, y))
+    Qbar, T = get_results_mice_imputation(Xy, y, num_imputations)
+    return Qbar, T
+
+
+# Now lets run all these imputation procedures.
+# We use the Boston dataset and analyze the outcomes of the beta coefficients
+# and their standard errors. We standardize the data before running the
+# procedure to be able to compare the coefficients. We run the procedure for
+# MCAR missingness only.
+#
+# Loading the data
+dataset = load_boston()
+X_full, y = dataset.data, dataset.target
+
+# Standardizing the data
+scaler = StandardScaler()
```

Comment:
I'll put this on top and it can be reused everywhere
```ESTIMATOR = Pipeline(steps=[('scaler', StandardScaler),
                            ('bayesianridge', BayesianRidge)])```